lib/backend/firestore/firestorebk.go

```diff
@@ -232,7 +232,7 @@ func newRecordFromDoc(doc *firestore.DocumentSnapshot) (*record, error) {
 			// Value was a string. This document could've been written by an older
 			// version of our code.
 			var rl legacyRecord
-			if doc.DataTo(&rl) != nil {
+			if err := doc.DataTo(&rl); err != nil {
```

Comment:
I think we might want to return the original unmarshal error (onto a regular key) here rather than the new one.